subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -641,7 +670,11 @@ public ModularitySpec getModularity() {
      */
     @Override
     protected JvmTestExecutionSpec createTestExecutionSpec() {
-        validateToolchainConfiguration();
+        return createSpec();
+    }
+
+    @VisibleForTesting
```

Comment:
I am not a big fan of all those `VisibleForTesting` methods. Is there a way to test this without exposing the specs?